src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+        for (Nucleotide base : baseQualities.keySet()) {
+            for (BaseQuality currentBaseQuality : baseQualities.get(base)) {
+                homRefLogLikelihood += getSafeguardedLogProbability(currentBaseQuality.getReadErrorProbability() / 3 +
+                        currentBaseQuality.getMappingErrorProbability() / 4 +
+                        ((base == alleleRef) ? (1 - 4 * currentBaseQuality.getReadErrorProbability() / 3
+                                - currentBaseQuality.getMappingErrorProbability()) : 0));
+                homAltLogLikelihood += getSafeguardedLogProbability(currentBaseQuality.getReadErrorProbability() / 3 +
+                                currentBaseQuality.getMappingErrorProbability() / 4 +
+                                ((base == alleleAlt) ? (1 - 4 * currentBaseQuality.getReadErrorProbability() / 3
+                                        - currentBaseQuality.getMappingErrorProbability()) : 0));
+            }
+        }
+
+        /* return the sum of |hom,ref) and |hom,alt) likelihoods */
+        return GATKProtectedMathUtils.naturalLogSumExp(homRefLogLikelihood, homAltLogLikelihood);
+    }
+
+    /**
+     * [internal helper function]
+     * calculate the log likelihood of hetrozygosity from just alt and ref pileup for a given @alleleFraction
+     * (see CNV-method.pdf for details)
+     * @param alleleFraction ref-to-alt allele fraction
+     * @param alphaList [internal to getHetLogLikelihood]
+     * @param betaList [internal to getHetLogLikelihood]
+     * @return log likelihood
+     */
+    private double getRefAltHetLogLikelihoodFixedAlleleFraction(final double alleleFraction,
+                                                                final ArrayList<Double> alphaList,
+                                                                final ArrayList<Double> betaList) {
+        double logLikelihood = 0;
+        for (int i=0; i<alphaList.size(); i++) {
+            logLikelihood += getSafeguardedLogProbability(alphaList.get(i) + alleleFraction * betaList.get(i));
+        }
+
+        return logLikelihood;
+    }
+
+    /**
+     * calculate the log likelihood of a SNP site being heterozygous for a given read pileup
+     * (see CNV-method.pdf for details)
+     * @param baseQualities map of bases to list of their calling error probabilities at the SNP site
+     * @param alleleRef the ref allele base
+     * @param alleleAlt the alt allele base
+     * @return the log likelihood
+     */
+    @VisibleForTesting
+    public double getHetLogLikelihood(final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities,
+                                       final Nucleotide alleleRef, final Nucleotide alleleAlt) {
+
+        double errorLogLikelihood = 0.0;
+        double refAltLogLikelihood;
+        ArrayList<Double> alphaList = new ArrayList<>();
+        ArrayList<Double> betaList = new ArrayList<>();
+
+        for (Nucleotide base : baseQualities.keySet()) {
```

Comment:
Same, base can be final, basQualities should be tested for nulls if this method is public.